tests/test_graph.py

```diff
@@ -166,9 +166,8 @@ def test_serialize_graph_to_json(clear_default_graph, branching_graph):
 
 def test_serialize_graph_to_pickle(clear_default_graph, branching_graph):
     serialized = branching_graph.to_pickle()
-    deserialized = Graph.from_pickle(serialized).to_pickle()
-
-    assert serialized == deserialized
+    deserialized = Graph.from_pickle(serialized)
+    assert deserialized.to_json() == branching_graph.to_json()
```

Comment:
In python 3.8, the first time pickling, unpickling and pickling again seems to add some additional information to the object's serialized structure. However, the graph one obtains after unpickling is a perfect copy of the one before pickling.

The json serialization was the closest thing to an equality operation on graphs I could find.